firebase-firestore/src/androidMain/kotlin/dev/gitlive/firebase/firestore/Timestamp.kt

```diff
@@ -1,8 +1,33 @@
 package dev.gitlive.firebase.firestore
 
-actual typealias Timestamp = com.google.firebase.Timestamp
+import kotlinx.serialization.Serializable
 
-actual fun timestampNow(): Timestamp = Timestamp.now()
-actual fun timestampWith(seconds: Long, nanoseconds: Int) = Timestamp(seconds, nanoseconds)
-actual val Timestamp.seconds: Long get() = seconds
-actual val Timestamp.nanoseconds: Int get() = nanoseconds
+/** A class representing a platform specific Firebase Timestamp. */
+actual typealias PlatformTimestamp = com.google.firebase.Timestamp
+
+/** A base class that could be used to combine [Timestamp] and [Timestamp.ServerTimestamp] in the same field. */
+@Serializable(with = BaseTimestampSerializer::class)
+actual sealed class BaseTimestamp
```

Comment:
Interesting find. I have to use expect / actual sealed class because kotlin common and JS specific code are treated as different modules. this could be caused by the fact JS doesn't support packages.
unfortunately android studio check warns you to about exhaustive `when`s

Reply:
Thats weird...